include/sys/_types.h

```diff
@@ -128,7 +132,9 @@ typedef unsigned short __nlink_t;
 typedef long        __suseconds_t;  /* microseconds (signed) */
 typedef unsigned long   __useconds_t;   /* microseconds (unsigned) */
 
-#if __NEWLIB__ > 3
+#if __NEWLIB__ >= 4
```

Comment:
This one might be a bit of a problem where we have to detect on minor version too... After all, the declaration was for > 3, not >= 3. So, it would only be detecting the same thing as your >= 4.